### PR TITLE
Use new preprocessing for legacy MO

### PR DESCRIPTION
**Pull request by mvafin**
Root cause analysis: N/A

Solution: Switch to new preprocessing

Ticket: 76250

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names
* [x]  Transformation preserves tensor names

Validation:
* [x]  Unit tests - covered by existing tests
* [x]  Framework operation tests N/A
* [x]  Transformation tests N/A
* [x]  Model Optimizer IR Reader check

Documentation:
* [x]  Supported frameworks operations list
* [x]  Guide on how to convert the **public** model
* [x]  User guide update

### CODE REVIEW

**Comment by GlebKazantaev** [comment]
Setting "do not merge" label due to dependency on other tickets/changes

**Comment by mvafin** [comment]
PR needs to be merged with the corresponding commit to tests repo.

**Comment by mvafin** [comment]
mo_tests failure is expected and will be fixed by tests change, please see ticket for details.

**Comment by mvafin** [comment]
Removing do not merge. Changes can be merged with tests changes now

**Review by jane-intel — COMMENTED** [review]
Do I understand it correctly that we keep old RIC propagation only for TF case with Unpack -> Pack which is translated to RIC? Can we create a transformation that will do the same in c++ and get rid of all RIC transformations in MO?

**Comment by mvafin** [comment]
> Also I see mo_arg_test related to RIC failed -- is it expected?

Yes, it is expected. We need to also update tests, recommit pass if tests updated, see ticket for details.

> Do I understand it correctly that we keep old RIC propagation only for TF case with Unpack -> Pack which is translated to RIC? Can we create a transformation that will do the same in c++ and get rid of all RIC transformations in MO?

Yes, you are right. Pack->Unpack case becomes RIC in MO. I guess we can remove it, but I don't think this is in scope of this PR.

**Comment by rkazants** [comment]
> > Also I see mo_arg_test related to RIC failed -- is it expected?
> 
> Yes, it is expected. We need to also update tests, recommit pass if tests updated, see ticket for details.
> 
> > Do I understand it correctly that we keep old RIC propagation only for TF case with Unpack -> Pack which is translated to RIC? Can we create a transformation that will do the same in c++ and get rid of all RIC transformations in MO?
> 
> Yes, you are right. Pack->Unpack case becomes RIC in MO. I guess we can remove it, but I don't think this is in scope of this PR.

@mvafin, can you please create a ticket for moving RIC transformations to (MOC/common) offline transformations.

File: tools/mo/openvino/tools/mo/back/ReverseInputChannels.py [hunk 1]
```diff
@@ -556,8 +490,6 @@ def find_and_replace_pattern(self, graph: Graph):
         """
         Following transformations should run in strict order, that is why we disabled them all and run here 
         """
-        if graph.graph['cmd_params'].reverse_input_channels:
-            InsertReverseChannels().find_and_replace_pattern(graph)
         ReverseChannelsPropagationDown().find_and_replace_pattern(graph)
         ReverseChannelsPropagationUp().find_and_replace_pattern(graph)
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/back/ReverseInputChannels.py:494]: I see no code in MO now that will create `ReverseChannels` operation if you remove `InsertReverseChannels` transformation. It looks that these transformation become useless here.

**Review by rkazants — COMMENTED** [review]
Left one comment about RIC transformation in MO

File: tools/mo/openvino/tools/mo/back/ReverseInputChannels.py (hunk 1 above)
Comment (mvafin) [reply tools/mo/openvino/tools/mo/back/ReverseInputChannels.py:494]: Here it is https://github.com/openvinotoolkit/openvino/blob/65ac9b5036e744359b7b07e7651904dc209ddae5/tools/mo/openvino/tools/mo/front/tf/UnpackPackReverseInputChannels.py#L67

**Review by rkazants — APPROVED** [review]

**Comment by mvafin** [comment]
> > > Also I see mo_arg_test related to RIC failed -- is it expected?
> > 
> > 
> > Yes, it is expected. We need to also update tests, recommit pass if tests updated, see ticket for details.
> > > Do I understand it correctly that we keep old RIC propagation only for TF case with Unpack -> Pack which is translated to RIC? Can we create a transformation that will do the same in c++ and get rid of all RIC transformations in MO?
> > 
> > 
> > Yes, you are right. Pack->Unpack case becomes RIC in MO. I guess we can remove it, but I don't think this is in scope of this PR.
> 
> @mvafin, can you please create a ticket for moving RIC transformations to (MOC/common) offline transformations.

Done, ticket 89861

**Review by jane-intel — APPROVED** [review]

**Comment by jane-intel** [comment]
@mvafin please collaborate with CI to merge both PRs -- to mo_tests and to openvino

**Pull request merged**